diff --git a/ucm2/sof-hda-dsp/HiFi.conf b/ucm2/sof-hda-dsp/HiFi.conf
index 60a8693..4782346 100644
--- a/ucm2/sof-hda-dsp/HiFi.conf
+++ b/ucm2/sof-hda-dsp/HiFi.conf
@@ -87,33 +87,33 @@ If.monomic {
 	}
 	After.SectionDevice "Mic"
 	True {
-		SectionDevice."Headphones2" {
-			Comment "Headphones Stereo Microphone"
-
-			ConflictingDevice [
-				"Headset"
-			]
+		SectionDevice."Headset" {
+			Comment "Headset Mono Microphone"
 
 			EnableSequence [
-				cset "name='Input Source' 'Headphone Mic'"
+				cset "name='Input Source' Headset Mic"
 			]
 
 			Value {
-				CapturePriority 200
+				CapturePriority 300
 				<sof-hda-dsp/HDA-Capture-value.conf>
 				JackControl "Headphone Mic Jack"
 			}
 		}
 
-		SectionDevice."Headset" {
-			Comment "Headset Mono Microphone"
+		SectionDevice."Headphones2" {
+			Comment "Headphones Stereo Microphone"
+
+			ConflictingDevice [
+				"Headset"
+			]
 
 			EnableSequence [
-				cset "name='Input Source' Headset Mic"
+				cset "name='Input Source' 'Headphone Mic'"
 			]
 
 			Value {
-				CapturePriority 300
+				CapturePriority 200
 				<sof-hda-dsp/HDA-Capture-value.conf>
 				JackControl "Headphone Mic Jack"
 			}
