diff --git a/src/backends/meta-monitor.c b/src/backends/meta-monitor.c
index 00502bb..e5dfda2 100644
--- a/src/backends/meta-monitor.c
+++ b/src/backends/meta-monitor.c
@@ -1828,7 +1828,7 @@ meta_monitor_calculate_crtc_pos (MetaMonitor          *monitor,
  * The largest mainstream laptops have screens ~17in, and HiDPI external
  * monitors start at ~23in, so 20in is a good boundary point
  */
-#define UI_SCALE_LARGE_TARGET_DPI 110
+#define UI_SCALE_LARGE_TARGET_DPI 75
 #define UI_SCALE_LARGE_MIN_SIZE_INCHES 20
 
 static float
