diff --git a/extension.js b/extension.js
index ecefc93..47c805b 100644
--- a/extension.js
+++ b/extension.js
@@ -30,9 +30,9 @@ export default class NoTitleBarWhenMaximizedExtension extends Extension {
     global.display.disconnect(this._createdConnection);
     global.window_manager.disconnect(this._changedConnection);
 
-    this._forEachWindow((window) => {
-      this._restore(window);
-    });
+    // this._forEachWindow((window) => {
+    //   this._restore(window);
+    // });
 
     // Reset all state
     this._xWindows = null;
diff --git a/metadata.json b/metadata.json
index 55faa32..0d1c533 100644
--- a/metadata.json
+++ b/metadata.json
@@ -7,7 +7,8 @@
     "46",
     "47",
     "48",
-    "49"
+    "49",
+    "50"
   ],
   "url": "https://github.com/alecdotninja/no-titlebar-when-maximized",
   "uuid": "no-titlebar-when-maximized@alec.ninja",
