diff --git a/extension.js b/extension.js
index abcdc2b..30233b6 100644
--- a/extension.js
+++ b/extension.js
@@ -30,9 +30,9 @@ export default class NoTitleBarWhenMaximizedExtension extends Extension {
     global.display.disconnect(this._createdConnection);
     global.window_manager.disconnect(this._changedConnection);
 
-    this._forEachWindow((window) => {
-      this._restore(window);
-    });
+    // this._forEachWindow((window) => {
+    //   this._restore(window);
+    // });
 
     // Reset all state
     this._xWindows = null;
diff --git a/metadata.json b/metadata.json
index 73dab4e..db7a065 100644
--- a/metadata.json
+++ b/metadata.json
@@ -4,7 +4,8 @@
   "name": "No Titlebar When Maximized",
   "shell-version": [
     "45",
-    "46"
+    "46",
+    "47"
   ],
   "url": "https://github.com/alecdotninja/no-titlebar-when-maximized",
   "uuid": "no-titlebar-when-maximized@alec.ninja",
