diff --git a/extension.js b/extension.js
index abcdc2b..30233b6 100644
--- a/extension.js
+++ b/extension.js
@@ -30,9 +30,9 @@ export default class NoTitleBarWhenMaximizedExtension extends Extension {
     global.display.disconnect(this._createdConnection);
     global.window_manager.disconnect(this._changedConnection);
 
-    this._forEachWindow((window) => {
-      this._restore(window);
-    });
+    // this._forEachWindow((window) => {
+    //   this._restore(window);
+    // });

