diff --git a/extension.js b/extension.js
index 158f47e..b3f3276 100644
--- a/extension.js
+++ b/extension.js
@@ -187,6 +187,7 @@ const ResourceMonitor = GObject.registerClass(
 
       if (typeof NM !== "undefined") {
         NM.Client.new_async(null, (client) => {
+          this._nm_client = client;
           client.connect(
             "active-connection-added",
             this._onActiveConnectionAdded.bind(this)
