diff --git a/extension.js b/extension.js
index 69ca8a8..ac12679 100644
--- a/extension.js
+++ b/extension.js
@@ -145,6 +145,7 @@ const ResourceMonitor = GObject.registerClass(
 
       if (typeof NM !== "undefined") {
         NM.Client.new_async(null, (client) => {
+          this._nm_client = client;
           client.connect(
             "active-connection-added",
             this._onActiveConnectionAdded.bind(this)
