diff --git a/extension.js b/extension.js
index 0d65e8b..bae7215 100644
--- a/extension.js
+++ b/extension.js
@@ -3396,7 +3396,7 @@ export default class ResourceMonitorExtension extends Extension {
 
     const index = {
       left: -1,
-      center: 0,
+      center: -1,
       right: 0,
     };
 
diff --git a/prefs.js b/prefs.js
index e87dd19..a485fb8 100644
--- a/prefs.js
+++ b/prefs.js
@@ -138,6 +138,20 @@ const ResourceMonitorBuilderScope = GObject.registerClass(
   }
 );
 
+function realpath(path) {
+    try {
+        const file = Gio.File.new_for_path(path);
+        const info = file.query_info("standard::*", Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS, null);
+        if (!info.get_is_symlink()) {
+            return path;
+        }
+        return Gio.File.new_for_commandline_arg_and_cwd(info.get_symlink_target(), file.get_parent().get_path()).get_path();
+    } catch (e) {
+        return path;
+    }
+};
+
+
 const ResourceMonitorPrefsWidget = GObject.registerClass(
   class ResourceMonitorPrefsWidget extends GObject.Object {
     _connectSpinButton(settings, settingsName, element) {
@@ -1141,16 +1155,24 @@ const ResourceMonitorPrefsWidget = GObject.registerClass(
       let disksArray = settings.get_strv(DISK_DEVICES_LIST);
 
       // Execute 'df' command and parse the output
-      this._executeCommand(["df", "-x", "squashfs", "-x", "tmpfs"])
+      this._executeCommand(["df", "-x", "squashfs", "-x", "tmpfs", '-x', 'devtmpfs'])
         .then((output) => {
           const lines = output.split("\n");
 
+          let fsf = new Set();
+
           // Parse each line of the command output, skipping the header line
           for (let i = 1; i < lines.length - 1; i++) {
             const entry = lines[i].trim().split(/\s+/);
-            const filesystem = entry[0];
+            const filesystem = realpath(entry[0]);
             const mountPoint = entry[5];
 
+            if (fsf.has(filesystem)) {
+              continue;
+            }
+
+            fsf.add(filesystem);
+
             // Initialize button states and display name
             let statsButton = false;
             let spaceButton = false;
