diff --git a/extension.js b/extension.js
index ffd09da..9043e31 100644
--- a/extension.js
+++ b/extension.js
@@ -3354,7 +3354,7 @@ export default class ResourceMonitorExtension extends Extension {
 
     const index = {
       left: -1,
-      center: 0,
+      center: -1,
       right: 0,
     };
 
diff --git a/prefs.js b/prefs.js
index ed176f4..cd5395f 100644
--- a/prefs.js
+++ b/prefs.js
@@ -119,6 +119,20 @@ const ResourceMonitorBuilderScope = GObject.registerClass(
   }
 );
 
+function realpath(path) {
+    try {
+        const file = Gio.File.new_for_path(path);
+        const info = file.query_info("standard::*", Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS, null);
+        if (!info.get_is_symlink()) {
+            return path;
+        }
+        return Gio.File.new_for_commandline_arg_and_cwd(info.get_symlink_target(), file.get_parent().get_path()).get_path();
+    } catch (e) {
+        return path;
+    }
+};
+
+
 const ResourceMonitorPrefsWidget = GObject.registerClass(
   class ResourceMonitorPrefsWidget extends GObject.Object {
     _connectSpinButton(settings, settingsName, element) {
@@ -710,18 +724,26 @@ const ResourceMonitorPrefsWidget = GObject.registerClass(
       // Get current disks settings
       let disksArray = settings.get_strv(DISK_DEVICES_LIST);
 
-      this._executeCommand(["df", "-x", "squashfs", "-x", "tmpfs"]).then(
+      this._executeCommand(["df", "-x", "squashfs", "-x", "tmpfs", '-x', 'devtmpfs']).then(
         (output) => {
           let lines = output.split("\n");
 
+          let fsf = new Set();
+
           // Excludes the first line of output
           for (let i = 1; i < lines.length - 1; i++) {
             let line = lines[i];
             let entry = line.trim().split(/\s+/);
 
-            let filesystem = entry[0];
+            let filesystem = realpath(entry[0]);
             let mountedOn = entry[5];
 
+            if (fsf.has(filesystem)) {
+              continue;
+            }
+
+            fsf.add(filesystem);
+
             let dStButton = false;
             let dSpButton = false;
 
