diff --git a/extension.js b/extension.js
index db83273..1a3e0a0 100644
--- a/extension.js
+++ b/extension.js
@@ -2879,7 +2879,7 @@ class Extension {
 
         const index = {
             left: -1,
-            center: 0,
+            center: -1,
             right: 0,
         };
 
diff --git a/prefs.js b/prefs.js
index 5fa2695..fe6063b 100644
--- a/prefs.js
+++ b/prefs.js
@@ -123,6 +123,20 @@ const ResourceMonitorBuilderScope = GObject.registerClass({
     }
 });
 
+function realpath(path) {
+    try {
+        const file = Gio.File.new_for_path(path);
+        const info = file.query_info("standard::*", Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS, null);
+        if (!info.get_is_symlink()) {
+            return path;
+        }
+        return Gio.File.new_for_commandline_arg_and_cwd(info.get_symlink_target(), file.get_parent().get_path()).get_path();
+    } catch (e) {
+        return path;
+    }
+};
+
+
 const ResourceMonitorPrefsWidget = GObject.registerClass(
     class ResourceMonitorPrefsWidget extends GObject.Object {
         _connectSpinButton(settings, settingsName, element) {
@@ -476,17 +490,25 @@ const ResourceMonitorPrefsWidget = GObject.registerClass(
             // Get current disks settings
             let disksArray = settings.get_strv(DISK_DEVICES_LIST);
 
-            this._executeCommand(['df', '-x', 'squashfs', '-x', 'tmpfs']).then(output => {
+            this._executeCommand(['df', '-x', 'squashfs', '-x', 'tmpfs', '-x', 'devtmpfs']).then(output => {
                 let lines = output.split('\n');
 
+                let fsf = new Set();
+
                 // Excludes the first line of output
                 for (let i = 1; i < lines.length - 1; i++) {
                     let line = lines[i];
                     let entry = line.trim().split(/\s+/);
 
-                    let filesystem = entry[0];
+                    let filesystem = realpath(entry[0]);
                     let mountedOn = entry[5];
 
+                    if (fsf.has(filesystem)) {
+                        continue;
+                    }
+
+                    fsf.add(filesystem);
+
                     let dStButton = false;
                     let dSpButton = false;
 
