diff --git a/extension.js b/extension.js
index 29cd8a3..3a6d5fa 100644
--- a/extension.js
+++ b/extension.js
@@ -2043,6 +2043,7 @@ const ResourceMonitor = GObject.registerClass(
         }
 
         _refreshCpuTemperatureValue() {
+            log("REFRESH THERMAL CPU")
             if (this._thermalCpuTemperatureDevicesList.length > 0) {
                 for (let i = 0; i < this._thermalCpuTemperatureDevicesList.length; i++) {
                     const element = this._thermalCpuTemperatureDevicesList[i];
@@ -2055,47 +2056,43 @@ const ResourceMonitor = GObject.registerClass(
                         continue;
                     }
 
-                    if (GLib.file_test(path, GLib.FileTest.EXISTS)) {
-                        this._loadFile(path).then(contents => {
-                            const value = parseInt(ByteArray.toString(contents));
+                    let temps = [];
 
-                            this._cpuTemperatures += value / 1000;
-                            this._cpuTemperaturesReads++;
+                    if (GLib.file_test(path, GLib.FileTest.EXISTS)) {
+                        temps.push(this._loadFile(path).then(contents => {
+                            return parseInt(ByteArray.toString(contents)) / 1000;
+                        }));
+                    }
+                    Promise.all(temps).then(values => {
+                        let temp = values.reduce((a, b) => a + b, 0)
 
-                            if (this._cpuTemperaturesReads >= this._thermalCpuTemperatureDevicesList.length) {
-                                // Temperatures Average
-                                let unit = "";
-                                this._cpuTemperatures /= this._cpuTemperaturesReads;
+                        // Temperatures Average
+                        let unit = "";
+                        this._cpuTemperatures = temp / values.length;
 
-                                switch (this._thermalTemperatureUnit) {
-                                    case 'f':
-                                        this._cpuTemperatures = (this._cpuTemperatures * 1.8) + 32;
-                                        unit = '°F';
+                        switch (this._thermalTemperatureUnit) {
+                            case 'f':
+                                this._cpuTemperatures = (this._cpuTemperatures * 1.8) + 32;
+                                unit = '°F';
 
-                                        break;
+                                break;
 
-                                    case 'c':
+                            case 'c':
 
-                                    default:
-                                        unit = '°C';
+                            default:
+                                unit = '°C';
 
-                                        break;
-                                }
+                                break;
+                        }
 
-                                if (this._decimalsStatus) {
-                                    this._cpuTemperatureValue.text = `[${this._cpuTemperatures.toFixed(1)}${unit}`;
-                                } else {
-                                    this._cpuTemperatureValue.text = `[${this._cpuTemperatures.toFixed(0)}${unit}`;
-                                }
+                        if (this._decimalsStatus) {
+                            this._cpuTemperatureValue.text = `[${this._cpuTemperatures.toFixed(1)}${unit}`;
+                        } else {
+                            this._cpuTemperatureValue.text = `[${this._cpuTemperatures.toFixed(0)}${unit}`;
+                        }
 
-                                this._cpuTemperatures = 0;
-                                this._cpuTemperaturesReads = 0;
-                            }
-                        });
-                    } else {
-                        this._cpuTemperatureValue.text = _('[Temperature Error');
-                        // this._cpuTemperatureUnit.text = '';
-                    }
+                        this._cpuTemperatures = 0;
+                    });
                 }
             } else {
                 this._cpuTemperatureValue.text = _('[--');
