diff --git a/extension.js b/extension.js
index bae7215..a289c10 100644
--- a/extension.js
+++ b/extension.js
@@ -323,49 +323,6 @@ const ResourceMonitor = GObject.registerClass(
         style_class: "system-status-icon",
       });
 
-      // Unit
-      this._cpuTemperatureUnit = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: "°C",
-      });
-      this._cpuTemperatureUnit.set_style("padding-left: 0.125em;");
-
-      this._cpuFrequencyUnit = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: "KHz",
-      });
-      this._cpuFrequencyUnit.set_style("padding-left: 0.125em;");
-
-      this._cpuUnit = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: "%",
-      });
-      this._cpuUnit.set_style("padding-left: 0.125em;");
-
-      this._ramUnit = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: this._ramUnitType ? "%" : "KB",
-      });
-      this._ramUnit.set_style("padding-left: 0.125em;");
-
-      this._swapUnit = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: this._swapUnitType ? "%" : "KB",
-      });
-      this._swapUnit.set_style("padding-left: 0.125em;");
-
-      this._ethUnit = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: "K",
-      });
-      this._ethUnit.set_style("padding-left: 0.125em;");
-
-      this._wlanUnit = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: "K",
-      });
-      this._wlanUnit.set_style("padding-left: 0.125em;");
-
       // Value
       this._cpuValue = new St.Label({
         y_align: Clutter.ActorAlign.CENTER,
@@ -472,15 +429,12 @@ const ResourceMonitor = GObject.registerClass(
               case "cpu":
                 this._box.add_child(this._cpuIcon);
                 this._box.add_child(this._cpuValue);
-                this._box.add_child(this._cpuUnit);
 
                 this._box.add_child(this._cpuTemperatureBracketStart);
                 this._box.add_child(this._cpuTemperatureValue);
-                this._box.add_child(this._cpuTemperatureUnit);
                 this._box.add_child(this._cpuTemperatureBracketEnd);
                 this._box.add_child(this._cpuFrequencyBracketStart);
                 this._box.add_child(this._cpuFrequencyValue);
-                this._box.add_child(this._cpuFrequencyUnit);
                 this._box.add_child(this._cpuFrequencyBracketEnd);
                 this._box.add_child(this._cpuLoadAverageBracketStart);
                 this._box.add_child(this._cpuLoadAverageValue);
@@ -491,14 +445,12 @@ const ResourceMonitor = GObject.registerClass(
               case "ram":
                 this._box.add_child(this._ramIcon);
                 this._box.add_child(this._ramValue);
-                this._box.add_child(this._ramUnit);
 
                 break;
 
               case "swap":
                 this._box.add_child(this._swapIcon);
                 this._box.add_child(this._swapValue);
-                this._box.add_child(this._swapUnit);
 
                 break;
 
@@ -517,14 +469,12 @@ const ResourceMonitor = GObject.registerClass(
               case "eth":
                 this._box.add_child(this._ethIcon);
                 this._box.add_child(this._ethValue);
-                this._box.add_child(this._ethUnit);
 
                 break;
 
               case "wlan":
                 this._box.add_child(this._wlanIcon);
                 this._box.add_child(this._wlanValue);
-                this._box.add_child(this._wlanUnit);
 
                 break;
 
@@ -548,15 +498,12 @@ const ResourceMonitor = GObject.registerClass(
             switch (element) {
               case "cpu":
                 this._box.add_child(this._cpuValue);
-                this._box.add_child(this._cpuUnit);
 
                 this._box.add_child(this._cpuTemperatureBracketStart);
                 this._box.add_child(this._cpuTemperatureValue);
-                this._box.add_child(this._cpuTemperatureUnit);
                 this._box.add_child(this._cpuTemperatureBracketEnd);
                 this._box.add_child(this._cpuFrequencyBracketStart);
                 this._box.add_child(this._cpuFrequencyValue);
-                this._box.add_child(this._cpuFrequencyUnit);
                 this._box.add_child(this._cpuFrequencyBracketEnd);
                 this._box.add_child(this._cpuLoadAverageBracketStart);
                 this._box.add_child(this._cpuLoadAverageValue);
@@ -567,14 +514,12 @@ const ResourceMonitor = GObject.registerClass(
 
               case "ram":
                 this._box.add_child(this._ramValue);
-                this._box.add_child(this._ramUnit);
                 this._box.add_child(this._ramIcon);
 
                 break;
 
               case "swap":
                 this._box.add_child(this._swapValue);
-                this._box.add_child(this._swapUnit);
                 this._box.add_child(this._swapIcon);
 
                 break;
@@ -593,14 +538,12 @@ const ResourceMonitor = GObject.registerClass(
 
               case "eth":
                 this._box.add_child(this._ethValue);
-                this._box.add_child(this._ethUnit);
                 this._box.add_child(this._ethIcon);
 
                 break;
 
               case "wlan":
                 this._box.add_child(this._wlanValue);
-                this._box.add_child(this._wlanUnit);
                 this._box.add_child(this._wlanIcon);
 
                 break;
@@ -1115,16 +1058,14 @@ const ResourceMonitor = GObject.registerClass(
         (this._netEthStatus && !this._netAutoHideStatus),
         true,
         this._ethIcon,
-        this._ethValue,
-        this._ethUnit
+        this._ethValue
       );
       this._basicItemStatus(
         (this._netWlanStatus && this._nmWlanStatus) ||
         (this._netWlanStatus && !this._netAutoHideStatus),
         true,
         this._wlanIcon,
-        this._wlanValue,
-        this._wlanUnit
+        this._wlanValue
       );
     }
 
@@ -1156,16 +1097,14 @@ const ResourceMonitor = GObject.registerClass(
         (this._netEthStatus && !this._netAutoHideStatus),
         true,
         this._ethIcon,
-        this._ethValue,
-        this._ethUnit
+        this._ethValue
       );
       this._basicItemStatus(
         (this._netWlanStatus && this._nmWlanStatus) ||
         (this._netWlanStatus && !this._netAutoHideStatus),
         true,
         this._wlanIcon,
-        this._wlanValue,
-        this._wlanUnit
+        this._wlanValue
       );
     }
 
@@ -1274,8 +1213,7 @@ const ResourceMonitor = GObject.registerClass(
         !this._cpuFrequencyStatus &&
         !this._cpuLoadAverageStatus,
         this._cpuIcon,
-        this._cpuValue,
-        this._cpuUnit
+        this._cpuValue
       );
     }
 
@@ -1304,7 +1242,6 @@ const ResourceMonitor = GObject.registerClass(
         !this._cpuLoadAverageStatus,
         this._cpuIcon,
         this._cpuFrequencyValue,
-        this._cpuFrequencyUnit,
         this._cpuFrequencyBracketStart,
         this._cpuFrequencyBracketEnd
       );
@@ -1380,8 +1317,7 @@ const ResourceMonitor = GObject.registerClass(
         this._ramStatus,
         true,
         this._ramIcon,
-        this._ramValue,
-        this._ramUnit
+        this._ramValue
       );
     }
 
@@ -1438,8 +1374,7 @@ const ResourceMonitor = GObject.registerClass(
         this._swapStatus,
         true,
         this._swapIcon,
-        this._swapValue,
-        this._swapUnit
+        this._swapValue
       );
     }
 
@@ -1651,16 +1586,14 @@ const ResourceMonitor = GObject.registerClass(
         (this._netEthStatus && !this._netAutoHideStatus),
         true,
         this._ethIcon,
-        this._ethValue,
-        this._ethUnit
+        this._ethValue
       );
       this._basicItemStatus(
         (this._netWlanStatus && this._nmWlanStatus) ||
         (this._netWlanStatus && !this._netAutoHideStatus),
         true,
         this._wlanIcon,
-        this._wlanValue,
-        this._wlanUnit
+        this._wlanValue
       );
     }
 
@@ -1694,8 +1627,7 @@ const ResourceMonitor = GObject.registerClass(
         (this._netEthStatus && !this._netAutoHideStatus),
         true,
         this._ethIcon,
-        this._ethValue,
-        this._ethUnit
+        this._ethValue
       );
     }
 
@@ -1724,8 +1656,7 @@ const ResourceMonitor = GObject.registerClass(
         (this._netWlanStatus && !this._netAutoHideStatus),
         true,
         this._wlanIcon,
-        this._wlanValue,
-        this._wlanUnit
+        this._wlanValue
       );
     }
 
@@ -1758,7 +1689,6 @@ const ResourceMonitor = GObject.registerClass(
         !this._cpuLoadAverageStatus,
         this._cpuIcon,
         this._cpuTemperatureValue,
-        this._cpuTemperatureUnit,
         this._cpuTemperatureBracketStart,
         this._cpuTemperatureBracketEnd
       );
@@ -2228,7 +2158,7 @@ const ResourceMonitor = GObject.registerClass(
           this._cpuValue.style = this._getUsageColor(cpuCurr, this._cpuColors);
 
           // Update CPU usage display with optional decimal precision
-          this._cpuValue.text = `${this._getValueFixed(cpuCurr)}`;
+          this._cpuValue.text = `${this._getValueFixed(cpuCurr)}%`;
         })
         .catch((error) =>
           console.error("[Resource_Monitor] Error reading cpu:", error)
@@ -2285,16 +2215,14 @@ const ResourceMonitor = GObject.registerClass(
           if (this._ramUnitType === "perc") {
             const percentValue = (100 * value) / total;
             this._ramValue.style = this._getUsageColor(percentValue, this._ramColors);
-            this._ramValue.text = `${this._getValueFixed(percentValue)}`;
-            this._ramUnit.text = "%";
+            this._ramValue.text = `${this._getValueFixed(percentValue)}%`;
           } else {
             [value, unit] = this._convertValueToUnit(
               value,
               this._ramUnitMeasure
             );
             this._ramValue.style = this._getUsageColor(value, this._ramColors);
-            this._ramValue.text = `${this._getValueFixed(value)}`;
-            this._ramUnit.text = unit;
+            this._ramValue.text = `${this._getValueFixed(value)}${unit}`;
           }
         })
         .catch((error) =>
@@ -2355,8 +2283,7 @@ const ResourceMonitor = GObject.registerClass(
               percentValue,
               this._swapColors
             );
-            this._swapValue.text = `${this._getValueFixed(percentValue)}`;
-            this._swapUnit.text = "%";
+            this._swapValue.text = `${this._getValueFixed(percentValue)}%`;
           } else {
             [value, unit] = this._convertValueToUnit(
               value,
@@ -2366,8 +2293,7 @@ const ResourceMonitor = GObject.registerClass(
               value,
               this._swapColors
             );
-            this._swapValue.text = `${this._getValueFixed(value)}`;
-            this._swapUnit.text = unit;
+            this._swapValue.text = `${this._getValueFixed(value)}${unit}`;
           }
         })
         .catch((error) =>
@@ -2409,8 +2335,7 @@ const ResourceMonitor = GObject.registerClass(
 
             this._diskStatsBox.update_element_value(
               filesystem,
-              `${this._getValueFixed(rw[0])}|${this._getValueFixed(rw[1])}`,
-              unit,
+              `${this._getValueFixed(rw[0])}|${this._getValueFixed(rw[1])}${unit}`,
               this._getUsageColor(rw, this._diskStatsColors)
             );
           };
@@ -2460,7 +2385,6 @@ const ResourceMonitor = GObject.registerClass(
                     this._diskStatsBox.update_element_value(
                       filesystem,
                       "--|--",
-                      "",
                       ""
                     );
                   }
@@ -2497,8 +2421,7 @@ const ResourceMonitor = GObject.registerClass(
 
               this._diskSpaceBox.update_element_value(
                 filesystem,
-                `${value}`,
-                "%",
+                `${value}%`,
                 this._getUsageColor(value, this._diskSpaceColors)
               );
             } else {
@@ -2515,8 +2438,7 @@ const ResourceMonitor = GObject.registerClass(
 
               this._diskSpaceBox.update_element_value(
                 filesystem,
-                `${this._getValueFixed(value)}`,
-                unit,
+                `${this._getValueFixed(value)}${unit}`,
                 this._getUsageColor(value, this._diskSpaceColors)
               );
             }
@@ -2534,6 +2456,7 @@ const ResourceMonitor = GObject.registerClass(
 
           let duTot = [0, 0];
           let du = [0, 0];
+          let unit = "";
 
           // Exclude the first two header lines
           for (let i = 2; i < lines.length - 1; i++) {
@@ -2569,7 +2492,7 @@ const ResourceMonitor = GObject.registerClass(
               this._netUnit === "bits"
             );
             du = result.values;
-            this._ethUnit.text = result.unit;
+            unit = result.unit;
           }
 
           // Set color based on thresholds
@@ -2578,7 +2501,7 @@ const ResourceMonitor = GObject.registerClass(
           // Display the download/upload values with appropriate decimal places
           this._ethValue.text = `${this._getValueFixed(
             du[0]
-          )}|${this._getValueFixed(du[1])}`;
+          )}|${this._getValueFixed(du[1])}${unit}`;
         })
         .catch((error) =>
           console.error("[Resource_Monitor] Error reading eth:", error)
@@ -2592,6 +2515,7 @@ const ResourceMonitor = GObject.registerClass(
 
           let duTot = [0, 0];
           let du = [0, 0];
+          let unit = "";
 
           // Exclude the first two header lines
           for (let i = 2; i < lines.length - 1; i++) {
@@ -2627,7 +2551,7 @@ const ResourceMonitor = GObject.registerClass(
               this._netUnit === "bits"
             );
             du = result.values;
-            this._wlanUnit.text = result.unit;
+            unit = result.unit;
           }
 
           // Set color based on thresholds
@@ -2636,7 +2560,7 @@ const ResourceMonitor = GObject.registerClass(
           // Display the download/upload values with appropriate decimal places
           this._wlanValue.text = `${this._getValueFixed(
             du[0]
-          )}|${this._getValueFixed(du[1])}`;
+          )}|${this._getValueFixed(du[1])}${unit}`;
         })
         .catch((error) =>
           console.error("[Resource_Monitor] Error reading wlan:", error)
@@ -2670,8 +2594,7 @@ const ResourceMonitor = GObject.registerClass(
             value,
             this._cpuFrequencyColors
           );
-          this._cpuFrequencyValue.text = `${this._getValueFixed(value)}`;
-          this._cpuFrequencyUnit.text = unit;
+          this._cpuFrequencyValue.text = `${this._getValueFixed(value)}${unit}`;
         })
         .catch((error) =>
           console.error("[Resource_Monitor] Error reading cpu frequency:", error)
@@ -2742,8 +2665,7 @@ const ResourceMonitor = GObject.registerClass(
 
                 this._cpuTemperatureValue.text = `${this._getValueFixed(
                   value
-                )}`;
-                this._cpuTemperatureUnit.text = unit;
+                )}${unit}`;
 
                 // Apply color based on thresholds
                 this._cpuTemperatureValue.style = this._getUsageColor(
@@ -2799,8 +2721,7 @@ const ResourceMonitor = GObject.registerClass(
             // GPU utilization
             this._gpuBox.update_element_value(
               uuid,
-              `${usage}`,
-              "%",
+              `${usage}%{unit}`,
               this._getUsageColor(usage, this._gpuColors)
             );
 
@@ -2824,8 +2745,7 @@ const ResourceMonitor = GObject.registerClass(
               const percentValue = (100 * memoryValue) / memoryTotal;
               this._gpuBox.update_element_memory_value(
                 uuid,
-                `${this._getValueFixed(percentValue)}`,
-                "%",
+                `${this._getValueFixed(percentValue)}%`,
                 this._getUsageColor(percentValue, this._gpuMemoryColors)
               );
             } else {
@@ -2836,8 +2756,7 @@ const ResourceMonitor = GObject.registerClass(
 
               this._gpuBox.update_element_memory_value(
                 uuid,
-                `${this._getValueFixed(memoryValue)}`,
-                memoryUnit,
+                `${this._getValueFixed(memoryValue)}${memoryUnit}`,
                 this._getUsageColor(memoryValue, this._gpuMemoryColors)
               );
             }
@@ -2848,8 +2767,7 @@ const ResourceMonitor = GObject.registerClass(
 
             this._gpuBox.update_element_thermal_value(
               uuid,
-              `${this._getValueFixed(tempValue)}`,
-              tempUnit,
+              `${this._getValueFixed(tempValue)}${tempUnit}`,
               this._getUsageColor(tempValue, this._thermalGpuColors)
             );
           });
@@ -2962,7 +2880,6 @@ const DiskContainer = GObject.registerClass(
       this._elementsPath = [];
       this._elementsName = [];
       this._elementsValue = [];
-      this._elementsUnit = [];
     }
 
     set_element_width(width) {
@@ -2996,7 +2913,6 @@ const DiskContainer = GObject.registerClass(
       this._elementsPath = [];
       this._elementsName = [];
       this._elementsValue = [];
-      this._elementsUnit = [];
 
       this.remove_all_children();
     }
@@ -3025,14 +2941,7 @@ const DiskContainerStats = GObject.registerClass(
         x_align: Clutter.ActorAlign.END,
       });
 
-      this._elementsUnit["single"] = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: "K",
-      });
-      this._elementsUnit["single"].set_style("padding-left: 0.125em;");
-
       this.add_child(this._elementsValue["single"]);
-      this.add_child(this._elementsUnit["single"]);
 
       this.idleOld["single"] = 0;
       this.rwTotOld["single"] = [0, 0];
@@ -3054,15 +2963,8 @@ const DiskContainerStats = GObject.registerClass(
         x_align: Clutter.ActorAlign.END,
       });
 
-      this._elementsUnit[filesystem] = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: "K",
-      });
-      this._elementsUnit[filesystem].set_style("padding-left: 0.125em;");
-
       this.add_child(this._elementsName[filesystem]);
       this.add_child(this._elementsValue[filesystem]);
-      this.add_child(this._elementsUnit[filesystem]);
 
       this.idleOld[filesystem] = 0;
       this.rwTotOld[filesystem] = [0, 0];
@@ -3075,10 +2977,8 @@ const DiskContainerStats = GObject.registerClass(
             if (element !== "single") {
               this._elementsName[element].hide();
               this._elementsValue[element].hide();
-              this._elementsUnit[element].hide();
             } else {
               this._elementsValue[element].show();
-              this._elementsUnit[element].show();
             }
           });
 
@@ -3091,10 +2991,8 @@ const DiskContainerStats = GObject.registerClass(
             if (element !== "single") {
               this._elementsName[element].show();
               this._elementsValue[element].show();
-              this._elementsUnit[element].show();
             } else {
               this._elementsValue[element].hide();
-              this._elementsUnit[element].hide();
             }
           });
 
@@ -3122,11 +3020,10 @@ const DiskContainerStats = GObject.registerClass(
       this.rwTotOld[filesystem] = rwTot;
     }
 
-    update_element_value(filesystem, value, unit, style = "") {
+    update_element_value(filesystem, value, style = "") {
       if (this._elementsValue[filesystem]) {
         this._elementsValue[filesystem].text = value;
         this._elementsValue[filesystem].style = style;
-        this._elementsUnit[filesystem].text = unit;
       }
     }
   }
@@ -3150,22 +3047,14 @@ const DiskContainerSpace = GObject.registerClass(
         x_align: Clutter.ActorAlign.END,
       });
 
-      this._elementsUnit[filesystem] = new St.Label({
-        y_align: Clutter.ActorAlign.CENTER,
-        text: this._diskSpaceUnitType ? "%" : "KB",
-      });
-      this._elementsUnit[filesystem].set_style("padding-left: 0.125em;");
-
       this.add_child(this._elementsName[filesystem]);
       this.add_child(this._elementsValue[filesystem]);
-      this.add_child(this._elementsUnit[filesystem]);
     }
 
-    update_element_value(filesystem, value, unit, style = "") {
+    update_element_value(filesystem, value, style = "") {
       if (this._elementsValue[filesystem]) {
         this._elementsValue[filesystem].text = value;
         this._elementsValue[filesystem].style = style;
-        this._elementsUnit[filesystem].text = unit;
       }
     }
   }
@@ -3179,11 +3068,8 @@ const GpuContainer = GObject.registerClass(
       this._elementsUuid = [];
       this._elementsName = [];
       this._elementsValue = [];
-      this._elementsUnit = [];
       this._elementsMemoryValue = [];
-      this._elementsMemoryUnit = [];
       this._elementsThermalValue = [];
-      this._elementsThermalUnit = [];
     }
 
     set_element_width(width) {
@@ -3239,11 +3125,8 @@ const GpuContainer = GObject.registerClass(
       this._elementsUuid = [];
       this._elementsName = [];
       this._elementsValue = [];
-      this._elementsUnit = [];
       this._elementsMemoryValue = [];
-      this._elementsMemoryUnit = [];
       this._elementsThermalValue = [];
-      this._elementsThermalUnit = [];
 
       this.remove_all_children();
     }
@@ -3269,12 +3152,6 @@ const GpuContainer = GObject.registerClass(
           x_align: Clutter.ActorAlign.END,
         });
 
-        this._elementsUnit[uuid] = new St.Label({
-          y_align: Clutter.ActorAlign.CENTER,
-          text: "%",
-        });
-        this._elementsUnit[uuid].set_style("padding-left: 0.125em;");
-
         this.add_child(
           new St.Label({
             y_align: Clutter.ActorAlign.CENTER,
@@ -3282,7 +3159,6 @@ const GpuContainer = GObject.registerClass(
           })
         );
         this.add_child(this._elementsValue[uuid]);
-        this.add_child(this._elementsUnit[uuid]);
         this.add_child(
           new St.Label({
             y_align: Clutter.ActorAlign.CENTER,
@@ -3301,12 +3177,6 @@ const GpuContainer = GObject.registerClass(
           x_align: Clutter.ActorAlign.END,
         });
 
-        this._elementsMemoryUnit[uuid] = new St.Label({
-          y_align: Clutter.ActorAlign.CENTER,
-          text: this._gpuMemoryUnitType ? "%" : "KB",
-        });
-        this._elementsMemoryUnit[uuid].set_style("padding-left: 0.125em;");
-
         this.add_child(
           new St.Label({
             y_align: Clutter.ActorAlign.CENTER,
@@ -3314,7 +3184,6 @@ const GpuContainer = GObject.registerClass(
           })
         );
         this.add_child(this._elementsMemoryValue[uuid]);
-        this.add_child(this._elementsMemoryUnit[uuid]);
         this.add_child(
           new St.Label({
             y_align: Clutter.ActorAlign.CENTER,
@@ -3333,12 +3202,6 @@ const GpuContainer = GObject.registerClass(
           x_align: Clutter.ActorAlign.END,
         });
 
-        this._elementsThermalUnit[uuid] = new St.Label({
-          y_align: Clutter.ActorAlign.CENTER,
-          text: "°C",
-        });
-        this._elementsThermalUnit[uuid].set_style("padding-left: 0.125em;");
-
         this.add_child(
           new St.Label({
             y_align: Clutter.ActorAlign.CENTER,
@@ -3346,7 +3209,6 @@ const GpuContainer = GObject.registerClass(
           })
         );
         this.add_child(this._elementsThermalValue[uuid]);
-        this.add_child(this._elementsThermalUnit[uuid]);
         this.add_child(
           new St.Label({
             y_align: Clutter.ActorAlign.CENTER,
@@ -3356,27 +3218,24 @@ const GpuContainer = GObject.registerClass(
       }
     }
 
-    update_element_value(uuid, value, unit, style = "") {
+    update_element_value(uuid, value, style = "") {
       if (this._elementsValue[uuid]) {
         this._elementsValue[uuid].text = value;
         this._elementsValue[uuid].style = style;
-        this._elementsUnit[uuid].text = unit;
       }
     }
 
-    update_element_memory_value(uuid, value, unit, style = "") {
+    update_element_memory_value(uuid, value, style = "") {
       if (this._elementsMemoryValue[uuid]) {
         this._elementsMemoryValue[uuid].text = value;
         this._elementsMemoryValue[uuid].style = style;
-        this._elementsMemoryUnit[uuid].text = unit;
       }
     }
 
-    update_element_thermal_value(uuid, value, unit, style = "") {
+    update_element_thermal_value(uuid, value, style = "") {
       if (this._elementsThermalValue[uuid]) {
         this._elementsThermalValue[uuid].text = value;
         this._elementsThermalValue[uuid].style = style;
-        this._elementsThermalUnit[uuid].text = unit;
       }
     }
   }
